test/integration/node/lifecycle_test.go

```diff
@@ -297,3 +303,29 @@ func TestTaintBasedEvictions(t *testing.T) {
 		})
 	}
 }
+
+// waitForPodCondition waits a pods to be matched to the given condition.
+func waitForPodCondition(c clientset.Interface, ns, podName, desc string, timeout time.Duration, condition podCondition, t *testing.T) error {
```

Comment:
nit: This function name is duplicated with test.waitForPodCondition which is passed to the above `desc string`.
That is a little less code readability, but that is not matter actually.
Maybe it is better to rename test.waitForPodCondition to the other one with different PR.

Reply:
Sure, I will follow up.